playbooks/openstack/openshift-cluster/build_image.yml

```diff
@@ -0,0 +1,177 @@
+
+
+- name: Launch image build instance
+  hosts: localhost
+  connection: local
+  gather_facts: yes
+  tasks:
+  # NOTE: we create a temporary network, subnet, router and security groups
+  # to have a known environment to launch the image build VM in.
+  # They get deleted once we save the image.
+  # TODO(shadower): allow specifying an existing subnet etc. instead.
+  - name: Create a network
+    os_network:
+      name: "{{ build_prefix }}-network"
+    register: network
+
+  - name: Create a subnet
+    os_subnet:
+      name: "{{ build_prefix }}-subnet"
+      network_name: "{{ network.network.name }}"
+      cidr: 192.168.23.0/24
+    register: subnet
+
+  - name: Create the router
+    os_router:
+      name: "{{ build_prefix }}-router"
+      network: "{{ openshift_openstack_external_network_name }}"
+      interfaces:
+      - "{{ subnet.id }}"
+    register: router
+
+  - name: Create a security group
+    os_security_group:
+      name: "{{ build_prefix }}-security-group"
+      description: Security group for the image build server
+    register: security_group
+
+  - name: Allow pinging the server
+    os_security_group_rule:
+      security_group: "{{ security_group.id }}"
+      protocol: icmp
+      port_range_min: -1
+      port_range_max: -1
+
+  - name: Allow SSH access
+    os_security_group_rule:
+      security_group: "{{ security_group.id }}"
+      protocol: tcp
+      port_range_min: 22
+      port_range_max: 22
+
+  - name: Launch the image build instance
+    os_server:
+      name: "{{ build_prefix }}-image-server"
+      network: "{{ network.id }}"
+      auto_ip: yes
+      flavor: "{{ openshift_openstack_default_flavor }}"
+      image: "{{ openshift_openstack_build_base_image }}"
+      key_name: "{{ openshift_openstack_keypair_name }}"
+      security_groups:
+      - "{{ security_group.id }}"
+      # Create a known SSH user so we can log in to the VM.
+      # TODO(shadower): should we create a temporary keypair & user here
+      # and delete it when done?
+      userdata: |
+        #cloud-config
+        system_info:
+          default_user:
+            name: {{ ansible_user|default('openshift') }}
+            sudo: ["ALL=(ALL) NOPASSWD: ALL"]
+        write_files:
+          - path: /etc/sudoers.d/00-openshift-no-requiretty
+            permissions: 440
+            content: |
+              Defaults:{{ ansible_user|default('openshift') }} !requiretty
+      state: present
+    register: image_vm
+
+  - name: Add host to nodes
+    add_host:
+      name: "{{ image_vm.openstack.accessIPv4 }}"
+      groups: nodes,OSEv3
+      ansible_become: true
+
+  - name: Wait for instance to respond to SSH
+    wait_for:
+      delay: 1
+      host: "{{ image_vm.openstack.accessIPv4 }}"
+      port: 22
+      state: started
+      timeout: 120
+
+
+- name: Wait for full SSH connection
```

Comment:
why do we need both waiting for SSH tasks?

Reply:
I don't think we need them necessarily, but the GCP playbook does the same and that's what I based this from:

https://github.com/openshift/openshift-ansible/blob/master/playbooks/gcp/openshift-cluster/build_image.yml#L65-L78